crates/holochain/src/core/workflow/call_zome_workflow.rs

```diff
@@ -245,6 +249,15 @@ pub struct CallZomeWorkspace {
     pub meta_cache: MetadataBuf,
 }
 
+/// When zome calls are made from
+/// other zome calls in the same cell
+/// we need to use an existing workspace
+/// and not write to it.
+pub enum CallZomeWorkspaceType {
```

Comment:
I think this needs a better name but nothing obvious is standing out to me

Reply:
Is it possible to normalize this to always take the lock by grabbing the lock before the first one too?